File: src/main/java/historywowa/global/infra/s3/presentation/application/S3Service.java
package historywowa.global.infra.s3.presentation.application;


import java.io.IOException;
import org.springframework.web.multipart.MultipartFile;

public interface S3Service {

    // 프로필 바꾸기
    String storeUserProFile(MultipartFile multipartFile, String userId) throws IOException;

    // 정적 지도 화면 이미지 저장
    String storeStaticMapImage(
        String key,
        byte[] bytes,
        String contentType
    );

}

File: src/main/kotlin/historywowa/global/infra/s3/presentation/application/S3Service.kt
package historywowa.global.infra.s3.presentation.application

import org.springframework.web.multipart.MultipartFile

interface S3Service {

    // 프로필 이미지 업로드
    fun storeImage(multipartFile: MultipartFile, userId: String): String?

    // 정적 지도 이미지 저장
    fun storeStaticMapImage(
            key: String,
            bytes: ByteArray,
            contentType: String
    ): String
}
